packages/foam-workspace-manager/src/core.ts

```diff
@@ -0,0 +1,89 @@
+import { Graph, Edge } from 'graphlib'
+
+
+type ID = string
+
+export interface FoamLink {
+  from: ID
+  to: ID
+  text: string
+}
+
+export interface OutLink {
+  to: ID
+  text: string
+}
+
+
+export class Bubble {
+  public id: ID
+  public title: string
+  public source: string
+  public path: string
+  public links: OutLink[]
+
+  constructor(id: ID, title: string, links: OutLink[], path: string, source: string) {
```

Comment:
Personally, for long argument lists made mainly of primitive data types, I prefer a named input object instead of positional arguments, since it's too easy to accidentally e.g. swap path and source order at call site

Reply:
I agree 100%.

In fact, for `Note`, I was thinking of just making it a type/interface, as I have removed all the methods I initially added here

Reply:
@riccardoferretti yes, all for using types over classes whenever we can!